### PR TITLE
Validate image path in integration exists and author image in pack exists

### PR DESCRIPTION
**Pull request by tcarmeli1**
## Description
Validate an integration has an image path and a pack has an author image path
## Related Issues:
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-9800 , https://jira-dc.paloaltonetworks.com/browse/CIAC-9797

### CODE REVIEW

File: demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Pack
+
+
+class AuthorImageExistsValidator(BaseValidator[ContentTypes]):
+    error_code = "IM109"
+    description = "Checks if the author image exists."
+    error_message = "author_image_doesn't_exists"
```
Comment (jbabazadeh) [inline demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py:19]:
```suggestion
    error_message = "Partners must provide a non-empty author image."
```

**Review by jbabazadeh — CHANGES_REQUESTED** [review]
Great work!! :) 
please see my comments

File: CommonServerUserPython.py
Comment (YuvHayun) [inline CommonServerUserPython.py:1]: ?

File: demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Integration
+
+
+class ImageExistsValidator(BaseValidator[ContentTypes]):
+    error_code = "IM100"
+    description = "Checks if the integration has an image path."
+    error_message = "You've created/modified a yml or package without providing an image as a .png file , please add an image in order to proceed."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py:19]: Let's also add the expected file path.

File: demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Pack
+
+class AuthorImageExistsValidator(BaseValidator[ContentTypes]):
+    error_code = "IM109"
+    description = "Checks if the pack has an author image path."
+    error_message = "Partner, You've created/modified a yml or package without providing an author image as a .png file , please add an image in order to proceed."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py:18]: Also here, let's add the path.

File: demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    related_field = "image"
+    is_auto_fixable = False
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py:20]:
```suggestion
    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
```

File: demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    related_field = "image"
+    is_auto_fixable = False
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py:21]:
```suggestion
    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Looks good!
Minor comments.

File: demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Integration
+
+
+class ImageExistsValidator(BaseValidator[ContentTypes]):
+    error_code = "IM100"
+    description = "Checks if the integration has an image path."
+    error_message = "You've created/modified a yml or package without providing an image as a .png file, please add an image"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py:19]:
```suggestion
    error_message = "You've created/modified a yml or package without providing an image as a .png file. Please make sure to add an image at {}."
```

File: demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    related_field = "image"
+    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
+    related_file_type = [RelatedFileType.IMAGE]
+
+    
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self,
+                message=f"{self.error_message} with the following path {content_item.name}_image.png in order to proceed.",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM100_image_exists_validation.py:29]:
```suggestion
                message=self.error_message.format(content_item.image_file),
```

File: demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Pack
+
+class AuthorImageExistsValidator(BaseValidator[ContentTypes]):
+    error_code = "IM109"
+    description = "Checks if the pack has an author image path."
+    error_message = "Partner, You've created/modified a yml or package without providing an author image as a .png file, please add an image with the following path Author_image.png in order to proceed."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py:18]:
```suggestion
    error_message = "You've created/modified a yml or package in a partner supported pack without providing an author image as a .png file. Please make sure to add an image at {}."
```

File: demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    related_field = "Author_image"
+    expected_git_statuses = [GitStatuses.ADDED, GitStatuses.MODIFIED]
+    related_file_type = [RelatedFileType.AUTHOR_IMAGE]
+
+    
+    def is_valid(self, content_items: Iterable[ContentTypes]) -> List[ValidationResult]:
+        return [
+            ValidationResult(
+                validator=self, message= self.error_message, content_object=content_item
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/IM_validators/IM109_author_image_exists_validation.py:27]:
```suggestion
                validator=self, message=self.error_message.format(content_item.author_image_path), content_object=content_item
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]

**Review by jbabazadeh — APPROVED** [review]
Great Work!

**Pull request merged**